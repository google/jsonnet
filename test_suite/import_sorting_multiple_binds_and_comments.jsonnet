local b = import "b.jsonnet",  # B comment.
      a = import "a.jsonnet";  # A comment.

local /* bbb */ b = import "b.jsonnet",  # B comment.
      /* aaa */ a = import "a.jsonnet";  # A comment.

local /* bbb
         bbb
         bbb */ b = import "b.jsonnet",  # B comment.
      /* aaa */ a = import "a.jsonnet";  # A comment.

// semi-ok...
local /* bbb */ b = import "b.jsonnet",  # B comment.
      /* aaa
         aaa
         aaa */ a = import "a.jsonnet";  # A comment.

// We don't really support that, it's going to be messy.
// There's an endline inserted before any paragraph comment.
local b = import "b.jsonnet",  /* BBB */ /* B comment.
      bbb
      bbb
      bbb */ a = import "a.jsonnet";  # A comment.

// The example above is equivalent to this one, due to fodder representation constraints.
local b = import "b.jsonnet",  /* BBB */
      /* B comment.
      bbb
      bbb
      bbb */
a = import "a.jsonnet";  # A comment.
true
