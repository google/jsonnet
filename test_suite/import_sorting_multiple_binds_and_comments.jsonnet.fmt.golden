local a = import "a.jsonnet";  # A comment.
local b = import "b.jsonnet";  # B comment.

local /* aaa */ a = import "a.jsonnet";  # A comment.
local /* bbb */ b = import "b.jsonnet";  # B comment.

local /* aaa */ a = import "a.jsonnet";  # A comment.
local
    /* bbb
       bbb
       bbb */
    b = import "b.jsonnet";  # B comment.

// semi-ok...
local
    /* aaa
       aaa
       aaa */
    a = import "a.jsonnet";  # A comment.
local /* bbb */ b = import "b.jsonnet";  # B comment.

// We don't really support that, it's going to be messy.
// There's an endline inserted before any paragraph comment.
local
    /* B comment.
    bbb
    bbb
    bbb */
    a = import "a.jsonnet";  # A comment.
local b = import "b.jsonnet"; /* BBB */

// The example above is equivalent to this one, due to fodder representation constraints.
local
    /* B comment.
    bbb
    bbb
    bbb */
    a = import "a.jsonnet";  # A comment.
local b = import "b.jsonnet"; /* BBB */
true
